BotRoyaleUnity/Assets/Scripts/Networking/SocketConnectionHandler.cs

```diff
@@ -126,10 +126,40 @@ public void EmitGameMessage(JSONObject data, Action<string> onMessageSent = null
         }
         else
         {
-            socket.Emit("game-message", data.ToString(), onMessageSent); ;
+            socket.Emit("game-message", data.ToString(), onMessageSent);
         }
     }
 
+    public void EmitCurrentParts(GameObject robot){
+        Vector2Int centerPos = robot.GetComponent<PartHandler>().GetCenterPos();
+        JSONObject data = new JSONObject();
+        data["action"] = "currentParts";
+        data["name"] = robot.name;
+        JSONArray parts = new JSONArray();
+        foreach(Transform part in robot.transform){
+            var partHealth = part.gameObject.GetComponent<PartHealth>();
+            JSONObject newPart = new JSONObject();
+            newPart["type"] = partHealth.GetPartType();
+            newPart["x"] = (partHealth.GetRelPos().x + centerPos.x);
+            newPart["y"] = (partHealth.GetRelPos().y + centerPos.y);
+            newPart["direction"] = partHealth.GetPartDirection();
+            newPart["health"] = partHealth.GetHealth();
+            parts.Add(newPart);
+        }
+        data["parts"] = parts;
+        socket.Emit("game-message", data.ToString());
+        
+    }
+
+    public void EmitEmptyParts(string name){
+        JSONObject data = new JSONObject();
+        data["action"] = "currentParts";
+        data["name"] = name;
+        JSONArray parts = new JSONArray();
```

Comment:
I'm a bit confused why the method EmitEmptyParts is needed since it appears to be very similar to EmitCurrentParts but assumes the robot has no parts

Reply:
This is so it can still be called even after the robot is destroyed.

Reply:
we can keep it like this for now, but I think the need for this method here is an indication that the robots dont have a good of handling what happens when they die